src/modules/languages/terraform.nix

```diff
@@ -13,9 +21,23 @@ in
       defaultText = lib.literalExpression "pkgs.terraform";
       description = "The Terraform package to use.";
     };
+
+    version = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+      default = null;
+      description = ''
+        The Terraform version to use.
+        This automatically sets the `languages.terraform.package` using [nixpkgs-terraform](https://github.com/stackbuilders/nixpkgs-terraform).
+      '';
+      example = "1.5.0 or 1.6.2";
+    };
   };
 
   config = lib.mkIf cfg.enable {
+    languages.terraform.package = lib.mkMerge [
+      (lib.mkIf (cfg.version != null) (nixpkgs-terraform.packages.${pkgs.stdenv.system}.${cfg.version} or (throw "Unsupported Terraform version, see https://github.com/stackbuilders/nixpkgs-terraform/blob/main/versions.json for the full list of supported versions.")))
```

Comment:
@oscar-izval maybe the version already exists in a new version of `nixpkgs-terraform`, in that sense, I think the first recommendation should be to try to upgrade the `nixpkgs-terraform` input first or look at `versions.json` for the full list of supported versions, alternatively. What do you think?